src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -159,6 +158,7 @@ public static void AddCommonServices(IServiceCollection services)
             services.TryAddSingleton<IDebugManager, DebugManager>();
             services.TryAddSingleton<IDebugStateProvider, DebugStateProvider>();
             services.TryAddSingleton<IEnvironment>(SystemEnvironment.Instance);
+            services.AddSingleton<IProxyMetadataManager, ProxyMetadataManager>();
```

Comment:
This guy had to be promoted to a "common" web app level service, since needed at a higher scope now

Reply:
Sounds good. The only comment I had was about the dependency lifetime for the options.

Reply:
fixed